### DISCUSSION TITLE
`readContracts` vs. `multicall` actions

### DISCUSSION THREAD

**Discussion by mds1** [wevm/wagmi#1652] [Q&A] [answered]
It looks like these are pretty much the same thing, and from what I can tell `readContracts` uses `multicall` under the hood. What is the difference between these and when should you use each?

Potentially related to this is that there's a `useContractReads` hook but no `useMulticall` hook. Is there a reason for this (and other) differences in what actions are available vs. what hooks are available?

**Comment by jxom** [comment, marked as answer]
`readContracts` internally falls back to individual reads if the chain does not support multicall3. 

We don't export a `useMulticall` hook as it is not guarenteed that all EVM chains have a multicall3 contract deployed, so `useContractReads` is a catch all hook.

**Reply by mds1** [reply]
Ahh awesome, thanks for the quick response!
